src/main/java/DilemmaDetector/Simulator/RigidBodyMapper.java

```diff
@@ -8,44 +8,12 @@
 public class RigidBodyMapper {
 
     //TODO get lane width from ontology? is it in centimeters
-    public static final int LANE_WIDTH = 300;
+    public static final int LANE_WIDTH = 300 / 100;
 
 
     public static List<Actor> createActors(Model model) {
         Map<Lane, ArrayList<Vehicle>> vehicleMap = model.getVehicles();
-
-        /*
-        iterowanie po mapie encji, żeby oddzielić pieszych od zwierząt i zrobić dwie tymczasowe
-        rozdzielone listy tak jak było do tej pory
-        Możliwe inne rozwiązanie dokonujące podziału dopiero przy tworzeniu RigidBody
-         */
-
-        Map<Lane, ArrayList<Animal>> animalMap = new HashMap<>(); // model.getAnimals();
-        for(Map.Entry<Lane, ArrayList<Living_entity>> entry : model.getEntities().entrySet()){
-            animalMap.put(entry.getKey(), new ArrayList<>());
-        }
-        for (Map.Entry<Lane, ArrayList<Living_entity>> entry : model.getEntities().entrySet()) {
-            for (Living_entity living_entity : entry.getValue()) {
-
-                if(living_entity instanceof Animal){
-                    ArrayList<Animal> currentAnimalList = animalMap.get(entry.getKey());
-                    currentAnimalList.add((Animal)living_entity);
-                }
-            }
-        }
-
-        Map<Lane, ArrayList<Pedestrian>> pedestrianMap = new HashMap<>(); //model.getPedestrians();
-        for(Map.Entry<Lane, ArrayList<Living_entity>> entry : model.getEntities().entrySet()){
-            pedestrianMap.put(entry.getKey(), new ArrayList<>());
-        }
-        for (Map.Entry<Lane, ArrayList<Living_entity>> entry : model.getEntities().entrySet()) {
-            for (Living_entity living_entity : entry.getValue()) {
-                if(living_entity instanceof Pedestrian){
-                    ArrayList<Pedestrian> currentPedestrianList = pedestrianMap.get(entry.getKey());
-                    currentPedestrianList.add((Pedestrian) living_entity);
-                }
-            }
-        }
+        Map<Lane, ArrayList<Living_entity>> livingEntityMap = model.getEntities();
```

Comment:
Może Entity a nie Living_Entity? Wyciągamy z modelu i tak Entity a do detekcji kolizji potrzebujemy też nieżywe twory. Takie rzutowanie w ogóle zadziała? W modelu można myślę też zmienić to na Entity.

Reply:
public Map<Lane, ArrayList<Living_entity>> getEntities() to sygnatura getEntities(), nie wiem czy tak łatwo jest zrzutować Mapę List Living_Entity na Mapę List Entity ale spróbuję